src/store/models/problem_sets.ts

```diff
@@ -0,0 +1,279 @@
+/* These are Problem Set interfaces */
+
+import { Dictionary, parseNonNegInt, Model, ParseError, generic,
+	InvalidFieldsException, ParseableModel, ModelField, parseParams } from '@/store/models/index';
+import { difference } from 'lodash';
+
+// const problem_set_types = [/hw/i, /quiz/i, /review/i];
+
+export enum ProblemSetType {
+	HW = 'HW',
+	QUIZ = 'QUIZ',
+	REVIEW_SET = 'REVIEW',
+	UNKNOWN = 'UNKNOWN'
+}
+
+export function parseProblemSetType(type: string) {
+	if (/hw/i.test(type)) {
+		return ProblemSetType.HW;
+	} else if (/quiz/i.test(type)) {
+		return ProblemSetType.QUIZ;
+	} else if (/review/i.test(type)) {
+		return ProblemSetType.REVIEW_SET;
+	}
+	throw new ParseError('ProblemSetType', `The problem set type '${type}' is not valid.`);
+}
+
+type ProblemSetParams = ParseableHWParams | ParseableQuizParams | ParseableReviewParams;
+type ProblemSetDates = ParseableHWDates | ParseableQuizDates | ParseableReviewDates;
+
+/* Problem Set (HomeworkSet, Quiz, ReviewSet ) classes */
+
+export interface ParseableProblemSet {
+	set_id?: string | number;
+	set_name?: string;
+	course_id?: string | number;
+	set_type?: string;
+	set_visible?: string | number | boolean;
+	set_params?: ProblemSetParams;
+	// set_params?: Dictionary<generic>;
+	set_dates?: ProblemSetDates;
+}
+
+export class ProblemSet extends Model(
+	['set_visible'], ['set_id', 'course_id'], ['set_type', 'set_name'],
+	['set_params', 'set_dates'],
+	{
+		set_type: { field_type: 'string', default_value: 'UNKNOWN' },
+		set_id: { field_type: 'non_neg_int', default_value: 0 },
+		set_name: { field_type: 'string' },
+		course_id: { field_type: 'non_neg_int', default_value: 0 },
+		set_visible: { field_type: 'boolean', default_value: false },
+	}) {
+	static REQUIRED_FIELDS = ['set_type'];
+	static OPTIONAL_FIELDS = ['set_id', 'set_name', 'course_id', 'set_visible'];
+
+	_date_fields: Array<string> = [];
+	_param_fields: ModelField = {};
+	set_params = {};
+	set_dates = {};
+
+	constructor(params: ParseableProblemSet = {}) {
+		super(params as ParseableModel);
+	}
+
+	isValid() {
+		throw 'You must override the isValid() method';
+	}
+
+	setDates(dates: Dictionary<generic> = {}){
+		// check that only valid dates are present
+		const invalid_dates = difference(Object.keys(dates), this._date_fields);
+		if (invalid_dates.length !== 0) {
+			throw new InvalidFieldsException('_all',
+				`The dates(s) '${invalid_dates.join(', ')}' are not valid for ${this.constructor.name}.`);
+		}
+
+		/* eslint-disable @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-member-access,
+		  @typescript-eslint/no-explicit-any */
+		Object.keys(dates).forEach(key => {
+			(this.set_dates as any)[key] = parseNonNegInt((dates as any)[key]);
+		});
+		// eslint-enable
```

Comment:
This also needs to be changed to `/* eslint-enable */`.  Sorry, I mean to mention that there were other instances of this in my last comment.  It seems this is the only other one though.

Reply:
theses are fixed.